Tella/Scenes/Camera/ViewModel/CameraViewModel.swift

```diff
@@ -84,14 +85,27 @@ class CameraViewModel: ObservableObject {
     
     func saveImage() {
         
-        guard let imageData = image?.fixedOrientation()?.pngData() else { return  }
-        guard let url = mainAppModel?.vaultManager.saveDataToTempFile(data: imageData, pathExtension: "png") else { return  }
+        guard let mainAppModel, let imageData else { return }
+        let isPreserveMetadataOn = mainAppModel.settings.preserveMetadata
         
-        saveFile(urlFile: url)
+        if currentLocation != nil && isPreserveMetadataOn {
+            let url = mainAppModel.vaultManager.createTempFileURL(pathExtension: FileExtension.heic.rawValue)
+            let isSaved = imageData.save(withLocation: currentLocation, fileURL: url)
+            if isSaved {
+                saveFile(urlFile: url)
+            }
+            
+        } else {
+            let url = mainAppModel.vaultManager.saveDataToTempFile(data: imageData,
```

Comment:
we are declaring multiple times url here. Why don't we just use an if let statement like this:

```
if let url = mainAppModel.vaultManager.saveDataToTempFile(data: imageData, pathExtension: FileExtension.heic.rawValue) {
        saveFile(urlFile: url)
    } else {
        debugLog("Failed to save image without location metadata.")
    }
    
```